### ISSUE TITLE
Add support to enforce disabled cops to be executed

### ISSUE THREAD

**Issue by wpolicarpo** [rubocop-hq/rubocop#6150]
According to the [documentation](http://docs.rubocop.org/en/latest/configuration/#disabling-cops-within-source-code), you can disable a cop by adding a comment to your source code:

```ruby
# rubocop:disable Metrics/LineLength, Style/StringLiterals
def foo
  bar
end
# rubocop:enable Metrics/LineLength, Style/StringLiterals
```
This is a useful feature indeed. Problem is that it enables people to ignore some easy to fix issues instead of actually fixing them.

It would be nice to have a way to enforce cops to be always evaluated, even if you disabled them within your source code.

The idea is to have a new configuration where you could disallow disabling cops using comments:

- Add new config `AllowSourceCodeDisabling` -  default is `true`
- Change cops to understand this new configuration
- Add new cop to check if you disabled a cop when it was not allowed to - or extend `Lint/UnneededCopDisableDirective` to do that

I would help with PRs if it's something that would be useful for other people.

**Comment by jaredbeck** [comment]
Hi Wanderson. 

My first thought was that OSS project maintainers will appreciate this. They've gone to the trouble of configuring their linter, it can be annoying when people submit PRs with `rubocop:disable` comments.

On the other hand, `rubocop:disable` comments give contributors a last-resort way to get something working, and some maintainers would rather make that first contribution easier, and then discuss linting afterward.

In an office environment, unless the team is very large, the lead dev could just talk to people and so probably doesn't need this setting.

In conclusion, I can see a use case for this, so :+1:.

> Add new config AllowSourceCodeDisabling - default is true
> Add new cop to check if you disabled a cop when it was not allowed to - or extend Lint/UnneededCopDisableDirective to do that

Instead of two new things, this could all be done with one cop. `CopsAreImmuneToComments` (j/k) I don't know what to call it, but it could be one cop.

**Comment by rrosenblum** [comment]
This sounds like it could be useful. The `Lint/UnneededCopDisableDirective` kind of does this already because if a cop is disabled via comment and there isn't actually an offense, `Lint/UnneededCopDisableDirective` will register an offense.

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.

**Comment by roooodcastro** [comment]
> Add new config AllowSourceCodeDisabling - default is true
> Add new cop to check if you disabled a cop when it was not allowed to - or extend Lint/UnneededCopDisableDirective to do that

> On the other hand, rubocop:disable comments give contributors a last-resort way to get something working, and some maintainers would rather make that first contribution easier, and then discuss linting afterward.

I also think this is useful, but maybe this could also be implemented as a one-off check? For example, still enable cops to be disabled, but have an option to run Rubocop once just to check all disabled cops. This still leaves Rubocop "contributor-friendly", but allows the maintainer to periodically check the project for too many disabled cops.

Btw, I'm willing to try and tackle this soon. WDYT?
